src/game/engine/cell.cpp

```diff
@@ -814,30 +814,36 @@ void CellClass::Draw_It(int x, int y, BOOL flag) const
             icon_num = Clear_Icon();
         }
 
-#if 0 // TODO Editor related stuff.
-        if ((g_InMapEditor && Map.field_8567) /*|| Debug_CellPassable*/) {
-            if (!Ground[Land].Is_Buildable() || Ground[Land].Get_Speed(SPEED_FOOT) == fixed_t::_0_1 && obj != nullptr) {
-                fading_table = DisplayClass::FadingRed;
-            }
-        }
-    
-        obj = Cell_Occupier();
-
-        if ((g_InMapEditor
-                && obj != nullptr /*&& obj->Get_Next() != nullptr*/) /*|| Debug_CellOccupied*/) {
-            if (Occupied_By_Foot()) {
-                fading_table = DisplayClass::FadingYellow;
+#ifdef CHRONOSHIFT_DEBUG
+        if (g_Debug_Passable) {
+            // editor version from edwin
+            if (g_InMapEditor) {
+                // impassable
+                if (Ground[Land].Get_Speed(SPEED_FOOT) == fixed_t(0, 0) || OccupierPtr != nullptr) {
+                    fading_table = DisplayClass::FadingRed;
+                }
+                // is occupied by multiple objects, could had been for finding overlapping objects?
+                if (OccupierPtr != nullptr && OccupierPtr->Get_Next() != nullptr) {
+                    fading_table = DisplayClass::FadingYellow;
+                }
+            // ingame version from RA beta
+            } else {
+                // impassable
+                if (Ground[Land].Get_Speed(SPEED_FOOT) == fixed_t(0, 0)
+                    || OccupierPtr != nullptr && OccupierPtr->What_Am_I() != RTTI_INFANTRY) {
+                    fading_table = DisplayClass::FadingRed;
+                // unit will be very slow on this cell, in a unmodified RA this won't ever trigger
+                } else if (Ground[Land].Get_Speed(SPEED_FOOT) <= fixed_t(0, 3)) {
```

Comment:
`fixed(0, 3)` will give you a result of 0, so you are right, it will never trigger. You probably mean `fixed(1, 3)` which will give you 0.33 ish.